sql/plan/sort.go

```diff
@@ -131,7 +131,7 @@ func (s *Sort) WithExpressions(exprs ...sql.Expression) (sql.Node, error) {
 }
 
 type sortIter struct {
-	s           *Sort
+	sortFields  sql.SortFields
```

Comment:
Getting rid of references to Nodes in these iter types is definitely a good change independent of my other comments about rolling sort / limit logic into union nodes